src/diffpy/utils/diffraction_objects.py

```diff
@@ -248,15 +248,27 @@ def _set_array_from_range(self, begin, end, step_size=None, n_steps=None):
             array = np.linspace(begin, end, n_steps)
         return array
 
-    def get_angle_index(self, angle):
-        count = 0
-        for i, target in enumerate(self.angles):
-            if angle == target:
+    def get_array_index(self, xtype, value):
```

Comment:
I am wondering if it is better to set default xtype to self.xtype, so that user only needs to enter the value when searching for an index. The xtype parameter would only need to be specified if the user wants to find the index of a different xarray.

Reply:
yes I like that.